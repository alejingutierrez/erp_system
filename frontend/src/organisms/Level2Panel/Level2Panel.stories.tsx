import type { Meta, StoryObj } from '@storybook/react';
import { Level2Panel } from './Level2Panel';

const meta: Meta<typeof Level2Panel> = {
  title: 'Organisms/Level2Panel',
  component: Level2Panel,
  tags: ['autodocs'],
  argTypes: {
    title: { control: 'text' },
    actions: { control: 'object' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    title: 'Productos',
    actions: [
      { label: 'Crear', onClick: () => alert('Crear') },
      { label: 'Importar', onClick: () => alert('Importar') },
      { label: 'Exportar', onClick: () => alert('Exportar') },
    ],
    children: 'Contenido del panel de nivel 2',
  },
};

export const WithoutActions: Story = {
  args: {
    title: 'Visi√≥n general',
    children: 'Contenido del panel de nivel 2 sin acciones',
  },
};
