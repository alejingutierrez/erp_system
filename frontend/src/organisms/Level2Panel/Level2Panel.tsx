import * as React from 'react';
import { cva, type VariantProps } from 'class-variance-authority';
import { cn } from '@/lib/utils';
import { Button } from '@/atoms/Button';
import { Heading } from '@/atoms/Heading';

const level2PanelVariants = cva('p-4 bg-background text-foreground', {
  variants: {},
  defaultVariants: {},
});

export interface Level2PanelProps
  extends React.HTMLAttributes<HTMLElement>,
    VariantProps<typeof level2PanelVariants> {
  title: string;
  actions?: {
    label: string;
    onClick: () => void;
  }[];
}

export const Level2Panel = React.forwardRef<HTMLElement, Level2PanelProps>(
  ({ className, title, actions, children, ...props }, ref) => {
    return (
      <div ref={ref} className={cn(level2PanelVariants(), className)} {...props}>
        <div className="flex justify-between items-center mb-4">
          <Heading level={2}>{title}</Heading>
          <div className="flex gap-2">
            {actions?.map((action) => (
              <Button key={action.label} onClick={action.onClick}>
                {action.label}
              </Button>
            ))}
          </div>
        </div>
        <div>{children}</div>
      </div>
    );
  },
);

Level2Panel.displayName = 'Level2Panel';
export { level2PanelVariants };
