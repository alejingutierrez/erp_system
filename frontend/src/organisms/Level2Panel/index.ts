export * from './Level2Panel';
