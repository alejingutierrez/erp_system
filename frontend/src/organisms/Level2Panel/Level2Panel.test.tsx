import { render, screen } from '@testing-library/react';
import { axe } from 'jest-axe';
import { Level2Panel } from './Level2Panel';

describe('Level2Panel', () => {
    it('should render correctly', () => {
        const { container } = render(<Level2Panel title="Test Panel" />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('should render with no accessibility violations', async () => {
        const { container } = render(<Level2Panel title="Test Panel" />);
        const results = await axe(container);
        expect(results).toHaveNoViolations();
    });

    it('should render title', () => {
        render(<Level2Panel title="Test Panel" />);
        expect(screen.getByText('Test Panel')).toBeInTheDocument();
    });

    it('should render actions', () => {
        const actions = [
            { label: 'Action 1', onClick: () => {} },
            { label: 'Action 2', onClick: () => {} },
        ];
        render(<Level2Panel title="Test Panel" actions={actions} />);
        expect(screen.getByText('Action 1')).toBeInTheDocument();
        expect(screen.getByText('Action 2')).toBeInTheDocument();
    });
});
