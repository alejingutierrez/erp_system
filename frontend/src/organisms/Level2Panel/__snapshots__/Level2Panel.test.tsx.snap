// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Level2Panel > should render correctly 1`] = `
<div
  class="p-4 bg-background text-foreground"
>
  <div
    class="flex justify-between items-center mb-4"
  >
    <h2
      class="font-heading text-4xl font-medium tracking-tight text-left text-primary"
    >
      Test Panel
    </h2>
    <div
      class="flex gap-2"
    />
  </div>
  <div />
</div>
`;
